crates/typst/src/layout/inline/collect.rs

```diff
@@ -201,7 +202,7 @@ pub fn collect<'a>(
                 );
                 let peeked = iter.peek().and_then(|(child, _)| {
                     if let Some(elem) = child.to_packed::<TextElem>() {
-                        elem.text().chars().next()
+                        elem.text().chars().find(|c| !is_directional_control(*c))
```

Comment:
could you explain when you use `is_directional_control` over `is_ignorable` / `is_exterior_to_quote`? I don't quite understand the difference.

Reply:
`is_directional_control` returns true for characters that have no effect on the smart quote algorithm – it’s as if they aren’t there in the first place. This is different from `is_ignorable`, which determines which characters lie immediately outside a quote, hence the renaming.

Reply:
Sounds like `is_ignorable` would be a good name for what `is_directional_control` is doing. I wonder whether we can apply the `Default_Ignorable` Unicode property to filter out more stuff. This is already used in paragraph layout.